src/evaluation/details.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashMap;
+
+#[derive(Clone, Default, Debug)]
+pub struct EvaluationDetails<T> {
+    pub flag_key: String,
+    pub value: T,
+    pub reason: Option<EvaluationReason>,
+    pub error: Option<EvaluationError>,
+    pub variant: Option<String>,
+    pub flag_metadata: FlagMetadata,
+}
+
+/// Eeason for evaluation.
```

Comment:
```suggestion
/// Reason for evaluation.
```

Reply:
I really appreciate your help of checking these small yet important details. :-)